### PR TITLE
Resource Filter Settings in URL

### PR DESCRIPTION
**Pull request by dmlb**
## Describe your changes
Query Params for search, tags, mode
- onMount for initial navigation
- $ for catching changes between URL and components
- replaceQuery HistoryAPI update to URL

## Related issue number/link
https://github.com/ClimateTown/knowledge-hub/issues/360

### CODE REVIEW

**Review by VeckoTheGecko — CHANGES_REQUESTED** [review]
Thanks for this feature. Questions about emojis in URL, and code quality/transparency items before shipping.

Happy to help with dev if you want me to push changes.

File: src/lib/components/FilterForm.svelte [hunk 1]
```diff
@@ -11,33 +11,58 @@
...
+  let filterOptions: FilterOption[]
+  $: ({ filterOptions } = filters)
 
   export let showFilterLogic: boolean = true
   // Whether all the selected tags must match the resource (vs any of the selected tags)
-  export let filterLogicAnd: boolean = true
+  let filterLogicAndCtrl: boolean = filters?.filterLogicAnd ?? true
   let filterLogic: FilterLogic
-  $: filterLogic = filterLogicAnd ? "and" : "or"
+  $: filterLogic = filterLogicAndCtrl ? "and" : "or"
```
[the remark this answers is not in the document]
Comment (dmlb) [reply src/lib/components/FilterForm.svelte:37]: you'd have to run the check against "and"/"or" to get a boolean for the HMTL attribute 🤔

File: src/routes/resources/+page.svelte [hunk 2]
```diff
@@ -11,28 +12,36 @@
...
+    filterByQuery,
+  } from "$lib/utils"
   import Search from "$lib/components/Search.svelte"
   import ListItem from "./ListItem.svelte"
   import ResourceNav from "$lib/components/ResourceNav.svelte"
   import ScrollTopButton from "$lib/components/ScrollTopButton.svelte"
   import FilterForm from "$lib/components/FilterForm.svelte"
   export let data: PageData
 
+  let query: string | null
```
[the remark this answers is not in the document]
Comment (dmlb) [reply src/routes/resources/+page.svelte:28]: in the URL param also?

[the remark this answers is not in the document]
Comment (VeckoTheGecko) [reply src/routes/resources/+page.svelte:28]: I think leave `q` in the URL param (I think thats common practice from what I've seen?) :)

File: src/lib/components/FilterForm.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (VeckoTheGecko) [reply src/lib/components/FilterForm.svelte:37]: Yeah thats a fair point. No avoiding it I guess, and nitpicky at that point 😅. All good how it is

**Review by VeckoTheGecko — APPROVED** [review]
LGTM now! Happy to merge

**Pull request merged**